xbmc/addons/gui/GUIDialogAddonInfo.cpp

```diff
@@ -610,15 +610,17 @@ void CGUIDialogAddonInfo::OnEnableDisable()
     if (PromptIfDependency(24075, 24091))
       return; //required. can't disable
 
-    CServiceBroker::GetAddonMgr().DisableAddon(m_localAddon->ID(), AddonDisabledReason::USER);
+    if (CServiceBroker::GetAddonMgr().DisableAddon(m_localAddon->ID(), AddonDisabledReason::USER))
+      m_item->SetProperty("Addon.Status", g_localizeStrings.Get(24023)); // Disabled
   }
   else
   {
     // Check user want to enable if lifecycle not normal
     if (!ADDON::GUI::CHelpers::DialogAddonLifecycleUseAsk(m_localAddon))
       return;
 
-    CServiceBroker::GetAddonMgr().EnableAddon(m_localAddon->ID());
+    if (CServiceBroker::GetAddonMgr().EnableAddon(m_localAddon->ID()))
+      m_item->SetProperty("Addon.Status", g_localizeStrings.Get(305)); // Enabled
   }
```

Comment:
What about instead, after the if statement, having a single statement:

```c++
bool enabled = CServiceBroker::GetAddonMgr().IsEnabled(m_localAddon->ID());
m_item->SetProperty("Addon.Status", enabled ?
    g_localizeStrings.Get(305) : // Enabled
    g_localizeStrings.Get(24023)); // Disabled
```

Because whatever the result of the block, we want the add-on status to be correct. Similar to the approach of "verify the invariant".

Reply:
Then good as is.